pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -1,40 +1,45 @@
 {
+  lib,
+  stdenv,
   bzip2,
   callPackage,
-  expat,
   fetchFromGitHub,
   fontconfig,
   freetype,
   glib,
   glslang,
   harfbuzz,
-  lib,
-  libadwaita,
   libGL,
-  libpng,
   libX11,
-  libXcursor,
-  libXi,
-  libXrandr,
+  libadwaita,
   ncurses,
   nixosTests,
   oniguruma,
   pandoc,
   pkg-config,
   removeReferencesTo,
-  stdenv,
   versionCheckHook,
   wrapGAppsHook4,
   zig_0_13,
-  zlib,
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/build.zig#L106
+  withAdwaita ? true,
```

Comment:
I also don't think that there's any value in letting the user build the package without Adwaita. It can be disabled at runtime if the user really needs to.

Reply:
It should also be noted that there are a number of developers that work on the GTK frontend that want to eliminate the possibility of building without Adwaita. There are many features that don't work at all without Adwaita (tab overviews is the big one) and other features have to be implemented twice, once for plain GTK, and once for GTK+Adwaita.

Reply:
This can be useful to do if you want to minimise your dependency tree. Though you'd be hard-pressed to get rid of Adwaita on any desktop system.

Still, use-cases where one doesn't need any fancy UI elements at all exist. The first thing I did when trying out ghostty was to disable the bar.

Reply:
> I also don't think that there's any value in letting the user build the package without Adwaita.

I think upstream may disagree with you there, at least for now. Why else would this be an option?

> It should also be noted that there are a number of developers that work on the GTK frontend that want to eliminate the possibility of building without Adwaita.

If they do it'll be easy to remove. I don't see much reason in hiding this option in the meantime